snapshot/snapshotter.go

```diff
@@ -135,3 +142,41 @@ func (cs *containerdSnapshotter) View(ctx context.Context, key, parent string, o
 	}
 	return cs.returnMounts(mf)
 }
+
+var redirectDirOption string
+var redirectDirOptionOnce sync.Once
+
+func getRedirectDirOption() string {
+	redirectDirOptionOnce.Do(func() {
+		if _, err := os.Stat("/sys/module/overlay/parameters/redirect_dir"); err != nil {
+			redirectDirOption = "" // redirect_dir unsupported
+			return
+		}
+		if userns.RunningInUserNS() {
+			redirectDirOption = "nofollow" // follow is not allowed in user ns
```

Comment:
Kernel before 4.15 seemed to lack this: https://github.com/torvalds/linux/commit/438c84c2f0c794f75ab55ce65c505b01bfce4480

While the upstream kernel didn't support overlayfs with userns at that time, Ubuntu XX.XX had supported it, so maybe we should just set `redirectDirOption=""`